catalog/ddms/catalog-ddms-transformer/src/main/java/org/codice/alliance/ddms/v2/Ddms20InputTransformer.java

```diff
@@ -0,0 +1,762 @@
+    metacard.setAttribute(getGeographicNames(ddms));
+    metacard.setAttribute(getGeographicRegions(ddms));
+    metacard.setAttribute(getBeNumbers(ddms));
+    metacard.setAttribute(getSubtitles(ddms));
+    // TODO: ext.ddms.icid-payload
+    // TODO: ext.ddms.security portion markings
+    // Portion-Markings: Title, Subtitle, description, contacts, related resources, security
+    // TODO: ext.ddms.related-resource not used?
+    // Todo: Not sure what ext.ddms.topic-type is for
+
+    return metacard;
+  }
+
+  private DdmsResource parseDdms(InputStream input) throws CatalogTransformerException {
+    try {
+      XMLStreamReader xmlStreamReader = XMLInputFactory.newFactory().createXMLStreamReader(input);
+      return new Ddms20XmlReader(xmlStreamReader).read();
+    } catch (XMLStreamException e) {
+      throw new CatalogTransformerException("Failed to parse DDMS 2.0", e);
+    }
+  }
+
+  private Attribute getTitle(DdmsResource ddms) {
+    String title = ddms.getTitles().stream().findFirst().map(Title::getValue).orElse(null);
+    return new AttributeImpl(Core.TITLE, title);
+  }
+
+  private Attribute getLocation(DdmsResource ddms) {
+    GeometryCollectionBuilder geometryCollectionBuilder = new GeometryCollectionBuilder();
+    for (GeospatialCoverage geospatialCoverage : ddms.getGeospatialCoverages()) {
+      for (BoundingBox boundingBox : geospatialCoverage.getBoundingBoxes()) {
+        geometryCollectionBuilder.polygon(
+            boundingBox.getWest(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getNorth(),
+            boundingBox.getEast(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getSouth(),
+            boundingBox.getWest(),
+            boundingBox.getNorth());
+      }
+
+      for (BoundingGeometry geometry : geospatialCoverage.getBoundingGeometries()) {
+        geometry
+            .getPolygons()
+            .stream()
+            .map(Polygon::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+
+        geometry
+            .getPoints()
+            .stream()
+            .map(Point::toString)
+            .map(this::convertGmlToWkt)
+            .filter(StringUtils::isNotBlank)
+            .forEach(geometryCollectionBuilder::addWkt);
+      }
+    }
+    String location = geometryCollectionBuilder.toString();
+    // can't return blank string or else the solar catalog provider will throw a
+    // NullPointerException
+    if (StringUtils.isBlank(location)) {
+      location = null;
+    }
+
+    return new AttributeImpl(Core.LOCATION, location);
+  }
+
+  private String convertGmlToWkt(String gmlXml) {
+    try {
+      return gml3ToWkt.convert(gmlXml);
+    } catch (ValidationException e) {
+      // TODO: Should we throw a catalog transform exception or log?
```

Comment:
We could allow it and mark the record INVALID with INVALID tag or throw exception. Either way we should address this TODO

Reply:
Agree with both - do not fail the entire transformation, and mark the record as INVALID.

Reply:
Well... I changed it to log the error and mark the metacard as invalid. Then I tried to test it, and couldn't find any gml input that would cause this exception but which the DDMS would successfully parse 😂 

I'll keep the change anyway. Please check it out to make sure it's what you were expecting.